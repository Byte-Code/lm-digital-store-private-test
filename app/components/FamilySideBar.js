import React, { Component, PropTypes } from 'react';
import ImmutablePropTypes from 'react-immutable-proptypes';
import { Link } from 'react-router';
import { Map } from 'immutable';
import styled from 'styled-components';

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  overflow: auto;
`;

const WorldName = styled.h3`
  font-family: LeroyMerlinSans Italic;
  font-size: 30px;
  width: 136px;
  line-height: 32px;
  margin: 46px 0 30px;
`;

const Family = styled.div`
  display: flex;
  flex-direction: column;
  width: 190px;
  height: 250px;
  box-shadow: 0 0 8px 0 #e4e4e4;
  margin-bottom: 20px;
`;

const ImageWrapper = styled.img`
  width: 190px;
  height: 190px;
`;

const FamilyName = styled.div`
  height: 60px;
  display: flex;
  justify-content: flex-start;
  align-items: center;
  &>p {
    text-transform: uppercase;
    padding-left: 10px;
    padding-right: 35px;
    font-family: LeroyMerlinSans Bold;
    font-size: 20px;
    line-height: 22px;
  }
`;

export default class FamilySideBar extends Component {
  static propTypes = {
    requestFetchWorld: PropTypes.func.isRequired,
    world: ImmutablePropTypes.map
  }

  static defaultProps = {
    world: Map()
  }

  componentDidMount() {
    this.props.requestFetchWorld();
  }

  renderFamilies() {
    const { world } = this.props;
    const families = world.get('families');

    return families.map(fam => {
      const familyName = fam.get('familyName');
      const image = fam.get('image');
      const categoryCode = fam.get('categoryCode');

      return (
        <Link
          key={familyName}
          to={`/catalogue/${categoryCode}`}
        >
          <Family>
            <ImageWrapper src={image} />
            <FamilyName>
              <p>{familyName}</p>
            </FamilyName>
          </Family>
        </Link>
      );
    });
  }

  render() {
    const { world } = this.props;

    if (world.isEmpty()) {
      return null;
    }

    const worldName = world.get('worldName');

    return (
      <Wrapper>
        <WorldName>{worldName}</WorldName>
        {this.renderFamilies()}
      </Wrapper>
    );
  }
}
