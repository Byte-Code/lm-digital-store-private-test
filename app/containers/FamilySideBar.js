import { connect } from 'react-redux';
import { push } from 'react-router-redux';

import FamilySideBar from '../components/FamilySideBar';
import { requestFetchWorld } from '../actions/worldActions';
import { getWorld } from '../reducers/selectors';

const mapStateToProps = (state) => ({
  world: getWorld(state)
});

const mapDispatchToProps = {
  requestFetchWorld,
  push
};

export default connect(mapStateToProps, mapDispatchToProps)(FamilySideBar);
