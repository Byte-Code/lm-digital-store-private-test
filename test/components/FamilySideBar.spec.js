import React from 'react';
import { fromJS } from 'immutable';
import { shallow, mount } from 'enzyme';

import FamilySideBar from '../../app/components/FamilySideBar';

describe('FamilySideBar', () => {
  const requestFetchWorld = jest.fn();
  const push = () => {};
  const closeMenu = () => {};

  it('should render null when world is empty', () => {
    const world = fromJS({});
    const result = shallow(
      <FamilySideBar
        world={world}
        push={push}
        closeMenu={closeMenu}
        requestFetchWorld={requestFetchWorld}
      />
    );
    expect(result).toMatchSnapshot();
  });

  it('should render properly when world is not empty', () => {
    const world = fromJS({
      families: [
        {
          familyName: 'name',
          image: 'image',
          categoryCode: 'code'
        }
      ]
    });
    const result = shallow(
      <FamilySideBar
        world={world}
        push={push}
        closeMenu={closeMenu}
        requestFetchWorld={requestFetchWorld}
      />
    );
    expect(result).toMatchSnapshot();
  });

  it('should call requestFetchWorld after mounting', () => {
    const world = fromJS({
      families: [
        {
          familyName: 'name',
          image: 'image',
          categoryCode: 'code'
        }
      ]
    });
    mount(
      <FamilySideBar
        world={world}
        push={push}
        closeMenu={closeMenu}
        requestFetchWorld={requestFetchWorld}
      />
    );
    expect(requestFetchWorld).toHaveBeenCalled();
  });
});
