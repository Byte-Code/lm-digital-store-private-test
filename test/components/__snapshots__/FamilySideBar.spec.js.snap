// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FamilySideBar should render null when world is empty 1`] = `null`;

exports[`FamilySideBar should render properly when world is not empty 1`] = `
<glamorous(div)>
  <glamorous(h3) />
  <glamorous(div)
    onClick={[Function]}
  >
    <glamorous(img)
      src="image"
    />
    <glamorous(div)>
      <p>
        name
      </p>
    </glamorous(div)>
  </glamorous(div)>
</glamorous(div)>
`;
